wordpress-mobile/WordPress-Android
# Issue/15632 moderate comments in reader posts

Fixes #15632

This PR adds ability to moderate comments in Reader posts for Admin and Editor users of the site that the post belongs to.

Comment moderation is performed after the Snackbar is dismissed. Because of the implementation of comment list, if you moderate a comment and immediately rotate the device, the comment will briefly reapear in the list. I worked on workaround, but the logic gets too complicated for such a rare usecase.

[![Image from Gyazo](https://i.gyazo.com/9a70ef5199e7d1432681c847581f3fef.gif)](https://gyazo.com/9a70ef5199e7d1432681c847581f3fef)

To test:

- Confirm that the moderation menu is visible only to the users with Administrator or Editor roles on the site the post belongs to.
- Confirm that comment is properly moderated.
- Confirm that the undo action works.
- Confirm that the empty view is visible after last comment was removed from the list.
- Confirm that share still works.

## Regression Notes
1. Potential unintended areas of impact
- Moderation menu is visible to users with incorrect roles.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing.

3. What automated tests I added (or what prevented me from doing so)
- Legacy code is very unfriendly to testing.

PR submission checklist:
